examples/browser/test/pages/main-page.ts

```diff
@@ -62,10 +69,102 @@ export class MainPage {
+        return browser.getLocation(`ul.p-MenuBar-content > .p-MenuBar-item:nth-child(${tabNumber}`, 'x');
     }
 
     public getxMenuPosition(): Number {
-        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x')
+        return this.driver.getLocation('.p-Widget.p-Menu.p-MenuBar-menu', 'x');
+    }
+
+    public isFolderStructureExpandable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        return browser.isExisting('span.theia-ExpansionToggle');
+    }
+
+    public isRightClickDirectoryAvailable(): Boolean {
+        console.log('*****************************' + browser.isExisting('li.p-TabBar-tab'));
+        browser.click('li.p-TabBar-tab');
+        browser.rightClick('#files.p-Widget.theia-Tree.theia-FileTree.theia-Files.p-StackedPanel-child', 50, 50);
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    public isRightClickInMainPanelAvailable(): Boolean {
+        browser.rightClick('#theia-main-dock-panel');
+        return browser.isVisible('.p-Widget.p-Menu');
+    }
+
+    public isDoubleClickOnFileAvailable(): Boolean {
+        browser.waitForExist('.theia-TreeNodeCaption');
+        browser.doubleClick('.theia-TreeNodeCaption');
+        return browser.isVisible('#theia-left-stack.p-Widget.p-Panel.p-StackedPanel.p-SplitPanel-child');
+    }
+
+    /*open is at position 1 in the context menu*/
+    public isContextMenuOpenAvailable(filePositionNumber: Number): Boolean {
+        browser.waitForVisible(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber}`);
+        browser.rightClick(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber})`, 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(1)');
+        return browser.isExisting('ul.p-TabBar-content');
+    }
+
+    /*open with is at position 2 in the context menu*/
+    public isContextMenuOpenWithAvailable(filePositionNumber: Number): Boolean {
+        browser.waitForVisible(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber}`);
+        browser.rightClick(`#files div.theia-TreeNode.theia-FileStatNode:nth-child(${filePositionNumber})`, 0, 0);
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(2)')
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(1)');
+        return browser.isExisting('ul.p-TabBar-content');
+    }
+
+    /*Delete is at position 10 in the context menu */
+    public isContextMenuNewFileAvailable(): Boolean {
+        var bool = false;
+        browser.rightClick('#theia-left-stack');
+        browser.waitForExist('div.p-Widget.p-Menu ul.p-Menu-content');
+        browser.click('ul li.p-Menu-item:nth-child(10)').pause(1000);
```

Comment:
the pause delay should be configurable... as a constant variable

Reply:
the position is also hardcoded here... we should make these positions part of the model as properties

Reply:
Actually we should not have isFoo() functions in the page model and rather have that in the tests and then use the page model to do suff like:

it('context menu is available) {
 browser.click(mainPage.ContenxtMenu().NewFile())
 browser.click(newFileDialog.button())
 elements = mainPage.getTreeRootElements();
....

see what I mean ?

That way the details of the page is abstracted into the mainPage model and the test can use something that is understandable to do it's tests.

Reply:
I will keep this for the next sprint. I will have to sit with Nazul and separate the tests depending on the use cases. Will take this in to account then. Is that ok?

Reply:
sure np